flink-connector-bigquery/src/test/java/com/google/cloud/flink/bigquery/sink/serializer/AvroToProtoSerializerTest.java

```diff
@@ -16,20 +16,731 @@
+
+        byte[] byteArray = "Any String you want".getBytes();
+        GenericRecord record =
+                new GenericRecordBuilder(avroSchema)
+                        .set("quantity", 1234)
+                        .set("fixed_field", new GenericData.Fixed(avroSchema, byteArray))
+                        .set("float_field", Float.parseFloat("12345.6789"))
+                        .set("enum_field", "C")
+                        .build();
+        DynamicMessage message = getDynamicMessageFromGenericRecord(record, descriptor);
+        assertEquals(1234, message.getField(descriptor.findFieldByNumber(1)));
+        assertEquals(
+                ByteString.copyFrom(byteArray), message.getField(descriptor.findFieldByNumber(2)));
+        assertEquals(12345.6789f, message.getField(descriptor.findFieldByNumber(3)));
+        assertEquals("C", message.getField(descriptor.findFieldByNumber(4)));
+    }
+
+    @Test
+    public void testNullablePrimitiveTypesConversion() throws BigQuerySerializationException {
+        BigQuerySchemaProvider bigQuerySchemaProvider =
+                TestBigQuerySchemas.getSchemaWithNullablePrimitiveTypes();
+        BigQueryProtoSerializer<GenericRecord> serializer = new AvroToProtoSerializer();
+        serializer.init(bigQuerySchemaProvider);
+        Schema avroSchema = bigQuerySchemaProvider.getAvroSchema();
+
+        GenericRecord record =
+                new GenericRecordBuilder(avroSchema)
+                        .set("number", null)
+                        .set("price", null)
+                        .set("species", null)
+                        .set("flighted", null)
+                        .set("sound", null)
+                        .set("required_record_field", null)
+                        .build();
+
+        ByteString byteString = serializer.serialize(record);
+        assertEquals("", byteString.toStringUtf8());
+    }
+
+    @Test
+    public void testUnionOfRemainingPrimitiveConversion() throws BigQuerySerializationException {
+        BigQuerySchemaProvider bigQuerySchemaProvider =
+                TestBigQuerySchemas.getSchemaWithUnionOfRemainingPrimitiveTypes();
+        BigQueryProtoSerializer<GenericRecord> serializer = new AvroToProtoSerializer();
+        serializer.init(bigQuerySchemaProvider);
+        Schema avroSchema = bigQuerySchemaProvider.getAvroSchema();
+
+        GenericRecord record =
+                new GenericRecordBuilder(avroSchema)
+                        .set("quantity", null)
+                        .set("fixed_field", null)
+                        .set("float_field", null)
+                        .set("enum_field", null)
+                        .build();
+        ByteString byteString = serializer.serialize(record);
+        assertEquals("", byteString.toStringUtf8());
+    }
+
+    // ------------ Test Schemas with Record of Different Types -----------
+    @Test
+    public void testRecordOfArrayConversation() {
```

Comment:
here and below: conversion instead of conversation :)

Reply:
Fixed.

Reply:
great; thanks! you can resolve this convo